docs/swarm/swarm-tutorial/index.md

```diff
@@ -40,41 +40,53 @@ To run this tutorial, you need the following:
+
+* [use Docker for Mac or Docker for Windows](#using-docker-for-mac-or-docker-for-windows)
+
+#### Install Docker Engine on Linux machines
+
+If you are using Linux based physical computers or cloud-provided computers as hosts, simply follow the [Linux install instructions](../../installation/index.md) for your platform. Spin up the three machines needed for this tutorial, and you are ready. You can test both single-node and multi-node swarm scenarios on Linux machines.
+
+#### Use Docker for Mac or Docker for Windows
+
+Alternatively, install the latest [Docker for Mac](/docker-for-mac/index.md) or [Docker for Windows](/docker-for-windows/index.md) application on a one computer. You can test both single-node and multi-node swarm from this computer, but you will need to use Docker Machine to test the multi-node scenarios.
+
+* You can use Docker for Mac or Windows to test _single-node_ features of swarm mode, including initializing a swarm with a single node, creating services, and scaling services. Docker "Moby" on Hyperkit (Mac) or Hyper-V (Windows) will serve as the single swarm node.
+
+* Currently, you cannot use Docker for Mac or Windows alone to test _multi-node_ swarm, but you can use the version of [Docker Machine](/machine/overview.md) that comes with it to create the swarm nodes locally or in the cloud, then follow the tutorial for all multi-node features. For this scenario, you will run commands from a Docker for Mac or Docker for Windows host, but that Docker host itself will _not_ be part of the swarm (that is; it will not be `manager1`, `worker1`, or `worker2` in our example). Once the nodes are created, all swarm commands will work as shown from the Mac terminal or Windows PowerShell with Docker for Mac or Docker for Windows running.
 
 ### The IP address of the manager machine
 
-The IP address must be assigned to an a network interface available to the host
-operating system. All nodes in the swarm must be able to access the manager at the IP address.
+The IP address must be assigned to a network interface available to the host operating system. All nodes in the swarm must be able to access the manager at the IP address.
 
-Because other nodes contact the manager node on its IP address, you should use a
-fixed IP address.
+Because other nodes contact the manager node on its IP address, you should use a fixed IP address.
 
->**Tip**: You can run `ifconfig` on Linux or Mac OS X to see a list of the
+You can run `ifconfig` on Linux or Mac OS X to see a list of the
 available network interfaces.
 
+If you are using Docker Machine, you can get the manager IP with either `docker-machine ls` or `docker-machine ip <MACHINE-NAME>` &#8212; for example, `docker-machine ip manager1`.
+
 The tutorial uses `manager1` : `192.168.99.100`.
 
 ### Open ports between the hosts
 
+The following ports must be available. Test initial commands first, as you may find that these ports are open by default.
```

Comment:
I don't understand the test instructions. Consider "On some systems, these ports are open by default."